External/Plugins/ASCompletion/Completion/ASGenerator.cs

```diff
@@ -2817,65 +2817,70 @@ private static void GenerateFunctionJob(GeneratorJobType job, ScintillaControl s
-                var parCount = 0;
-                var arrCount = 0;
-                var braCount = 0;
-                var genCount = 0;
-                var dquCount = 0;
-                var quoCount = 0;
-                var parameterIndex = 0;
-                var pos = wordStartPos;
-                while (pos-- > 0)
-                {
-                    var c = sci.CharAt(pos);
-                    if (c == ',' && parCount == 0 && arrCount == 0 && braCount == 0 && genCount == 0 && dquCount == 0 && quoCount == 0) parameterIndex++;
-                    else if (c == ']') arrCount++;
-                    else if (c == '[') arrCount--;
-                    else if (c == '}') braCount++;
-                    else if (c == '{') braCount--;
-                    else if (c == '>') genCount++;
-                    else if (c == '<') genCount--;
-                    else if (dquCount == 0 && c == '"' && sci.CharAt(pos - 1) != '\\') dquCount++;
-                    else if (dquCount > 0 && c == '"' && sci.CharAt(pos - 1) != '\\') dquCount--;
-                    else if (quoCount == 0 && c == '\'' && sci.CharAt(pos - 1) != '\\') quoCount++;
-                    else if (quoCount > 0 && c == '\'' && sci.CharAt(pos - 1) != '\\') quoCount--;
-                    else if (c == ')') parCount++;
-                    else if (c == '(' && dquCount == 0 && quoCount == 0)
+            ASResult callerExpr = null;
+            MemberModel caller = null;
+            var parCount = 0;
+            var arrCount = 0;
+            var braCount = 0;
+            var genCount = 0;
+            var dquCount = 0;
+            var quoCount = 0;
+            var parameterIndex = 0;
+            var pos = wordStartPos;
+            var endPos = member != null ? sci.PositionFromLine(member.LineFrom) : 0;
+            while (pos-- > endPos)
+            {
+                if (sci.PositionIsOnComment(pos)) continue;
+                var c = (char)sci.CharAt(pos);
+                if (c <= ' ') continue;
+                if (c == ',' && parCount == 0 && arrCount == 0 && braCount == 0 && genCount == 0 && dquCount == 0 && quoCount == 0) parameterIndex++;
+                else if (c == ';' && parCount == 0 && arrCount == 0 && braCount == 0 && genCount == 0 && dquCount == 0 && quoCount == 0) break;
+                else if (c == ']') arrCount++;
+                else if (c == '[') arrCount--;
+                else if (c == '}') braCount++;
+                else if (c == '{') braCount--;
+                else if (c == '>') genCount++;
+                else if (c == '<') genCount--;
+                else if (dquCount == 0 && c == '"' && sci.CharAt(pos - 1) != '\\') dquCount++;
+                else if (dquCount > 0 && c == '"' && sci.CharAt(pos - 1) != '\\') dquCount--;
+                else if (quoCount == 0 && c == '\'' && sci.CharAt(pos - 1) != '\\') quoCount++;
+                else if (quoCount > 0 && c == '\'' && sci.CharAt(pos - 1) != '\\') quoCount--;
```

Comment:
Wouldn't this behave unexpectedly when there is an escaped backslash before the quotes like `\\"`?
`ASComplete` has `IsEscapedCharacter(ScintillaControl sci, int position)` method which might be useful if so.